client/app/scripts/superdesk/directives/sdUserAvatar.js

```diff
@@ -14,16 +14,21 @@ define(['angular'], function (angular) {
                         element.hide();
                     });
 
-                    if (scope.src) {
-                        element.attr('src', scope.src).show();
-                        figure.addClass('no-bg');
-                    } else if (scope.initials) {
-                        var initials = scope.initials.replace(/\W*(\w)\w*/g, '$1').toUpperCase();
-                        element.hide().parent().html(initials);
-                        figure.addClass('initials');
-                    } else {
-                        element.hide();
-                        figure.removeClass('no-bg');
+                    scope.$watch('src', initAvatar);
```

Comment:
imo it would be better if you pass `user` to directive and there it can decide if it should render image or initials.
also with those 2 watchers it will run `initAvatar` at least 2x - so if you want to keep those params you can use `$watchGroup(['src', 'initials'], initAvatar)`